### PR TITLE
modules/py_display: Add user SPI LCD register control.

### PR DESCRIPTION
**Pull request by kwagyeman**
I added a stub to allow you to execute register reads and writes to the SPI display. I then implemented the SPI register write. I don't believe read will work for our driver as we didn't hookup that line.

Anyway, code is redundant from the DSI read/write. Thoughts what the API should be?

This fixes: https://github.com/openmv/openmv/issues/2066

### CODE REVIEW

**Comment by iabdalkader** [comment]
We can just have generic display.bus_read/write functions that call both. The SPI functions will just have to accept one extra unused arg (dcs) easier this way, or make bus_read/write accept va args. However I'm not sure this actually fixes anything, the SPI driver already writes some init registers, and it's hard-coded to support a specific controller's init/reset sequence. I don't think this is needed, unless it actually fixes that linked forums issue.

**Comment by kwagyeman** [comment]
I see, I was just trying to give the user an option to write their own custom register settings. But, I agree with going all the way and making a custom controller driver. That way then folks can submit their own drivers.

**Comment by iabdalkader** [comment]
This pull request has been mentioned on **OpenMV Forums**. There might be relevant details there:

https://forums.openmv.io/t/tft-lcd-screen-module/764/14

**Comment by kwagyeman** [comment]
I updated the dsi read/write to be generic bus read/write calls. 

Then, I added support for spi writing to the display regs via this generic interface. I also added controller support to the spi class.

However, I made it such that if you don't pass a controller, you get a default initialization. I did this not to break all code using the SPI LCD again as the previous change from the lcd module to the display module did.

Anyway, this PR should give folks an easy way to change the LCD settings.

**Comment by iabdalkader** [comment]
Don't you want to move the display controller to Python ?

File: src/omv/modules/py_spi_display.c [hunk 1]
```diff
@@ -394,6 +414,7 @@ STATIC const py_display_p_t py_display_p = {
     #ifdef OMV_SPI_DISPLAY_BL_PIN
     .set_backlight = spi_display_set_backlight,
     #endif
+    .bus_write = spi_write,
```
Comment (iabdalkader) [inline src/omv/modules/py_spi_display.c:417]: No `bus_read` ?

**Review by iabdalkader — APPROVED** [review]

File: src/omv/modules/py_spi_display.c (hunk 1 above)
Comment (kwagyeman) [reply src/omv/modules/py_spi_display.c:417]: We didn't connect it on the LCD shield so I don't have anything to test with at the moment. I can write a new display controller class for like a 320x240 screen and then test and add read support later.

**Comment by kwagyeman** [comment]
@iabdalkader - I'd like to leave the default one for our SPI shields in C as it makes the code work. Also, it's incredibly minimal. You can pass a display controller for new shields as mentioned above. I can produce that later.

**Pull request merged**